state.go

```diff
@@ -863,6 +863,13 @@ func (s *State) OnInterface(se *Session, i interface{}) (err error) {
 		}
 	case *MessageUpdate:
 		if s.MaxMessageCount != 0 {
+			var old *Message
+			old, err = s.Message(t.ChannelID, t.ID)
+			if err == nil {
+				oldCopy := *old
+				t.BeforeUpdate = &oldCopy
```

Comment:
`t.BeforeUpdate = old` should work here too, then you can get rid of oldCopy.

Reply:
No? Using old would cause it to use a copy of the pointer, which is pointing to a struct that is about to be updated. By doing this I'm creating a shallow copy of the struct.